nx/guides/advanced/automatic_differentiation.livemd

```diff
@@ -0,0 +1,182 @@
+# Automatic Differentation
+
+```elixir
+Mix.install([
+  {:nx, "~> 0.7"}
+])
+```
+
+## What is Function Differentiation?
+
+Nx, through the `Nx.Defn.grad/2` and `Nx.Defn.value_and_grad/3` functions allows the user to differentiate functions that were defined through `defn`.
```

Comment:
For what is worth, we don't explain in this section what is function differentiation. We do talk about differentiation, gradient and derivative, but we don't explain any of the terms. It may be valuable to give a step back and provide a high-level overview. And then either reduce the amount of terms or explicitly explain them.

Reply:
I added 2 paragraphs explaining derivatives and gradients